### PR TITLE
Add IPubFailure event handler

### PR DESCRIPTION
**Pull request by rbu**
@icemac We talked about this earlier during the sprint. Do you think this code makes sense?

Superseded by #9 to get a cleaner commit history.

### CODE REVIEW

File: src/Products/SiteErrorLog/SiteErrorLog.py [hunk 1]
```diff
@@ -329,3 +332,18 @@ def manage_addErrorLog(dispatcher, RESPONSE=None):
...
+        return
+
+    try:
+        error_log = aq_acquire(published, '__error_log__', containment=1)
+    except AttributeError:
+        pass
+    else:
+        error_log.raising(event.exc_info)
+
+provideHandler(_handle_wsgi_publisher_failure)
```
Comment (icemac) [inline src/Products/SiteErrorLog/SiteErrorLog.py]: Registering the handler should be done via ZCML in `configure.zcml` so it is registered in the correct registry.

See https://github.com/zopefoundation/zope.catalog/blob/ccbd0f444a0dd69d4fa244741215f43e9ef46648/src/zope/catalog/subscribers.zcml as an example.

File: src/Products/SiteErrorLog/SiteErrorLog.py [hunk 2]
```diff
@@ -329,3 +332,18 @@ def manage_addErrorLog(dispatcher, RESPONSE=None):
         RESPONSE.redirect(
             dispatcher.DestinationURL() +
             '/manage_main?manage_tabs_message=Error+Log+Added.')
+
+@adapter(IPubFailure)
+def _handle_wsgi_publisher_failure(event):
+    published = event.request.get('PUBLISHED')
```
Comment (icemac) [inline src/Products/SiteErrorLog/SiteErrorLog.py]: A docstring would be nice.

**Review by icemac — CHANGES_REQUESTED** [review]
Thank you for working on this PR. It mostly LGTM. Only the registration of the event handler should be moved to ZCML. This might require to change the test so it loads the ZCML. (Ask me for pointers how to do this if needed.)

Additionally a change log entry would be nice.

**Comment by rbu** [comment]
Fixed the points you raised, closing in favor of PR #9

**Pull request closed without merging**